repos/system_upgrade/el7toel8/actors/satelliteupgradedatamigration/actor.py

```diff
@@ -0,0 +1,36 @@
+import glob
+import os
+import shutil
+
+from leapp.actors import Actor
+from leapp.libraries.common.rpms import has_package
+from leapp.libraries.stdlib import run
+from leapp.models import InstalledRPM
+from leapp.tags import ApplicationsPhaseTag, IPUWorkflowTag
+
+POSTGRESQL_DATA_PATH = '/var/lib/pgsql/data/'
+POSTGRESQL_SCL_DATA_PATH = '/var/opt/rh/rh-postgresql12/lib/pgsql/data/'
+
+
+class SatelliteUpgradeDataMigration(Actor):
+    """
+    No documentation has been provided for the satellite_upgrade_data_migration actor.
+    """
+
+    name = 'satellite_upgrade_data_migration'
+    consumes = (InstalledRPM,)
+    produces = ()
+    tags = (IPUWorkflowTag, ApplicationsPhaseTag)
+
+    def process(self):
+        has_foreman = has_package(InstalledRPM, 'foreman') or has_package(InstalledRPM, 'foreman-proxy')
+        if not has_foreman:
+            return
+
+        # move PostgreSQL data to the new home
+        if os.path.exists(POSTGRESQL_DATA_PATH):
+            os.rmdir(POSTGRESQL_DATA_PATH)
+        if not os.path.exists('/var/lib/pgsql'):
```

Comment:
It's a bit odd you have a constant for `/var/lib/pgsql/data/` but here you hardcode the path. Should this get the parent from `POSTGRESQL_DATA_PATH`?

Reply:
in reality, this part is probably not needed anymore now that we properly upgrade SCL PostgreSQL to modular PostgreSQL